src/instruments/src/Common/hooks.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+//  Copyright (c) 2021 FlyByWire Simulations
+//  SPDX-License-Identifier: GPL-3.0
+
+import React, { useCallback } from 'react';
 import { getRootElement } from './defaults';
 
-export const useUpdate = (handler: (deltaTime: number) => void) => {
+/**
+ * Hook that requests a callback on every frame.
+ *
+ * This hook allows specifying dependencies through the `additionalDeps` parameter. However, you should probably not use that;
+ * if you are getting a SimVar value every frame through a hook, for example, it might be better to just retrieve the variable
+ * inside the callback. This avoids unnecessary re-renders of your component if that variable is only used in this callback.
+ *
+ * Only use dependencies if you absolutely must keep up with a component state in the callback.
+ *
+ * @param handler           a callback function, called every time MSFS fires an update event. This varies based on
+ *                          the "GLASS COCKPIT REFRESH RATE" sim setting, but takes deltaTime for stability.
+ * @param additionalDeps    an array of dependencies. this acts just like the same parameter on {@link useCallback}, which is
+ *                          actually what is used internally.
+ */
+export const useUpdate = (handler: (deltaTime: number) => void, additionalDeps: any[] = []) => {
```

Comment:
I like the new API, however this is not backwards-compatible:

Previously, a new handler function would have been used automatically because it would have been synced to the ref via useEffect. Now, if  additionalDeps is not passed, the value in useCallback will never be changed because the changes array is empty. While that is probably still useful, we will need to carefully inspect whether any previous usage of useUpdate passed a calback that depended on local variables (that would be stale after this change).

Reply:
I don't aim for this to be backwards compatible. The reason this worked before is not really because of the ref, it's because any component would re-render at the highest `useSimVar` refresh rate (often 30Hz) causing this handler to be synced via that ref.

I actually added this array because removing the frequent re-renders actually broke the old system, as explained above.